plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -73,6 +77,12 @@ type predicateMetadata struct {
 	podRequest                *schedulercache.Resource
 	podPorts                  map[int]bool
 	matchingAntiAffinityTerms []matchingPodAntiAffinityTerm
+
+	// CheckServiceAffinity: pods cache, otherwise this is a large query, will slow things down
+	// when there are 100s of services/pods.
+	lock                       sync.Mutex
+	servicePods                []*api.Pod
+	servicePodsLastRefreshTime time.Time
```

Comment:
I don't understand the logic behind refreshing it. Basically, metadata should be precoumputed for a given pod before starting computing predicates for each nodes and we should use exactly the same data for each nodes.